### DISCUSSION TITLE
Correct usage with Next.js Typescript?

### DISCUSSION THREAD

**Discussion by eduwass** [permafrost-dev/node-ray#79] [Q&A] [answered]
Hi there,

just getting to try this out so excuse me if I'm doing things wrong or it's just not supported, would appreciate any kind of help/guidance!

The thing is I am able to send some messages through to Ray, but a lot of information seems to be missing, e.g:

![CleanShot 2022-05-23 at 02 50 47](https://user-images.githubusercontent.com/1070495/169724325-32eb8d8f-7b73-48e8-abb8-27578ec72de6.png)

- Ray doesn't seem to know path, see: `unknown-file.js:1`
-  `caller()` and `trace()` functions do not output anything

I am using "node-ray/web" because if I try to import "node-ray" I'll get the following error

```
error - ./node_modules/node-ray/node_modules/locate-path/index.js:3:0
Module not found: Can't resolve 'fs'

Import trace for requested module:
./node_modules/node-ray/node_modules/find-up/index.js
./node_modules/node-ray/dist/index.esm.mjs
./pages/index.tsx
```

The project is based on create-next-app here's the package.json contents:

```
{
  "name": "project",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@emotion/react": "^11.9.0",
    "@emotion/styled": "^11.8.1",
    "@hookform/resolvers": "^2.8.10",
    "@mui/material": "^5.7.0",
    "@mui/x-data-grid": "^5.10.0",
    "axios": "^0.27.2",
    "cookies-next": "^2.0.4",
    "next": "12.1.6",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "react-hook-form": "^7.31.1",
    "yup": "^0.32.11"
  },
  "devDependencies": {
    "@types/node": "17.0.31",
    "@types/react": "18.0.9",
    "@types/react-dom": "18.0.3",
    "eslint": "8.15.0",
    "eslint-config-next": "12.1.6",
    "node-ray": "^1.18.0",
    "typescript": "4.6.4"
  }
}
```

Thanks in advance!

**Comment by patinthehat** [comment]
@eduwass Would you mind opening an Issue (instead of a discussion) for this issue? Thank you for bringing it to my attention! :+1:

**Reply by eduwass** [reply]
For sure, created here: https://github.com/permafrost-dev/node-ray/issues/84

Thanks 👍

**Comment by patinthehat** [comment, marked as answer]
@eduwass For use within Next.js, I recommend using the new [`react-ray`](https://github.com/permafrost-dev/react-ray) package, which is the official React integration for Ray.
